source/common/http/filter_manager.cc

```diff
@@ -960,6 +972,8 @@ void DownstreamFilterManager::sendLocalReply(
   ASSERT(!state_.under_on_local_reply_);
   const bool is_head_request = state_.is_head_request_;
   const bool is_grpc_request = state_.is_grpc_request_;
+  // Local reply closes the stream even if downstream is not half closed.
+  state_.force_close_stream_ = true;
```

Comment:
instead of setting a boolean here (which adds to the booleans we need to keep track of in all places, WDYT of adding a stopSending function which we call both on local reply and on receipt of error response, which stops downstream processing?

I think that'd then be reuseable for the "stream reset preceding data" bug as we could convert no_error stream resets into stop sending calls.

Reply:
Makes sense. I have reworked this a bit and added the stopDecoding method.